plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/CodeTemplatesTest.java

```diff
@@ -86,20 +97,34 @@ public void testMaterialize()
     
     IFolder webapp = main.getFolder("webapp");
     IFolder webinf = webapp.getFolder("WEB-INF");
-    IFile appengineWebXml = webinf.getFile("appengine-web.xml");
-    Assert.assertTrue(appengineWebXml.exists());
-    Document doc = buildDocument(appengineWebXml);
-    NodeList threadsafeElements = doc.getDocumentElement().getElementsByTagName("threadsafe");
-    Assert.assertEquals("Must have exactly one threadsafe", 1, threadsafeElements.getLength());
-    String threadsafe = threadsafeElements.item(0).getTextContent();
-    Assert.assertEquals("true", threadsafe);
-    NodeList sessionsEnabledElements 
-        = doc.getDocumentElement().getElementsByTagName("sessions-enabled");
-    Assert.assertEquals("Must have exactly one sessions-enabled", 
-        1, sessionsEnabledElements.getLength());
-    String sessionsEnabled = sessionsEnabledElements.item(0).getTextContent();
-    Assert.assertEquals("false", sessionsEnabled);
-    
+
+    if (isStandardProject) {
```

Comment:
It's best to avoid conditional logic in tests so it's clearer what runs and no assertions are accidentally skipped. Here it's probably easiest to not share helper methods.